docs/design_spec.md

```diff
@@ -0,0 +1,94 @@
+# Nimbess Agent Design
+
+## Overview
+
+Nimbess Agent's main purpose is to manage an external data plane. The Agent's main responsibilities
+include:
+
+ 1. Translating Northbound (NB) application-specific resources into network oriented resources across the
+ entire network fabric.
+ 2. Managing and serving as the control plane for the locally assigned, and external data plane (Southbound).
+ 
+ The Agent is distributed over every host and each Agent instance is capable of viewing and managing the
+ network configuration for the entire fabric. However, only the Agent local to a data plane (on the same host)
+ is capable of directly configuring that data plane.
+ 
+## Agent Northbound Design
+ 
+Nimbess Agent is able to support multiple container platforms and by default supports Kubernetes (K8S). It
+also supports the CNI spec for handling CNI requests from Kubelet or other container platforms which use
+CNI. The Nimbess Agent NB is responsible for translating K8S resources into network-oriented resources.
+
+The Agent exposes a gRPC based CNI API in order to service CNI requests. The requests are translated into
+Nimbess Ports, and a Network Pipeline is built for that Port. For other networking related container platform
+resources, such as a K8S Service, Ingress, etc; Nimbess Agent will handle translating those resources into their
+relevant network-oriented constructs and applied to one or more pipelines.
+ 
+## Network Constructs in Nimbess Agent
```

Comment:
Do you want to do ### and put these under ## Nimbess Agent

Reply:
no. This whole document is about the nimbess agent